### PR TITLE
Fix disappearing terrain and sky

### PR DESCRIPTION
**Pull request by bagnell**
* Fixed the sky atmosphere fragment shader that was causing the atmosphere to disappear when near or under the surface of the ellipsoid.
* Fixed occlusion culling to only test occlusion when the camera is not under the surface of the ellipsoid.

Fixes both #2271 and #2415.

### CODE REVIEW

**Comment by bagnell** [comment]
The easiest way to test is to set:

`scene.screenSpaceCameraController.enableCollisionDetection = false;`

and rotate around one of the 3D models positioned on the ground.

**Comment by pjcozzi** [comment]
I get this atmosphere on Mac in Chrome and Firefox:

image

Here it is without the globe:

image

**Comment by pjcozzi** [comment]
Looks OK near the ground:

image

**Comment by pjcozzi** [comment]
When we fix this, can we also add a comment to SkyAtmosphereFS.glsl about the workaround/fix?

**Comment by pjcozzi** [comment]
Can we also change the default for `screenSpaceCameraController.minimumZoomDistance` in this PR?  What do you think?  1m? 0.1m?

**Comment by pjcozzi** [comment]
Looks good.

The only other issue is zoom to the dead sea and quickly use the middle mouse button to take a horizon view - the camera will start moving backwards.  If this is not an easy fix, submit a new issue.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look.

The camera problem you mentioned is intentional. The camera slides over terrain when a collision is detected to prevent it from getting stuck.

**Comment by pjcozzi** [comment]
Do you get these test failures:
```
Scene/ScreenSpaceCameraController camera does go below the terrain in 3D when collision detection is disabled
Expected 1.000000000846144 to be less than 1.
Error: Expected 1.000000000846144 to be less than 1.
    at stack (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1475:17)
    at buildExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1445:14)
    at Spec.Env.expectationResultFactory (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:584:18)
    at Spec.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:332:34)
    at Expectation.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:528:21)
    at Expectation.toBeLessThan (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1399:12)
    at http://localhost:8080/Specs/Scene/ScreenSpaceCameraControllerSpec.js:1059:52
    at Object.<anonymous> (http://localhost:8080/Specs/spec-main.js:137:30)
    at attemptAsync (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1793:24)
    at QueueRunner.run (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1746:26)
Scene/ScreenSpaceCameraController camera does go below the terrain in CV when collision detection is disabled
Expected 1.000000000846144 to be less than 1.
Error: Expected 1.000000000846144 to be less than 1.
    at stack (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1475:17)
    at buildExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1445:14)
    at Spec.Env.expectationResultFactory (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:584:18)
    at Spec.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:332:34)
    at Expectation.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:528:21)
    at Expectation.toBeLessThan (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1399:12)
    at http://localhost:8080/Specs/Scene/ScreenSpaceCameraControllerSpec.js:1075:35
    at Object.<anonymous> (http://localhost:8080/Specs/spec-main.js:137:30)
    at attemptAsync (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1793:24)
    at QueueRunner.run (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1746:26)
Scene/ScreenSpaceCameraController camera does not go below the terrain in CV with the transform set
Expected 0.9999999948034711 to be greater than or equal to 1.
Error: Expected 0.9999999948034711 to be greater than or equal to 1.
    at stack (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1475:17)
    at buildExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1445:14)
    at Spec.Env.expectationResultFactory (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:584:18)
    at Spec.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:332:34)
    at Expectation.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:528:21)
    at Expectation.toBeGreaterThanOrEqualTo (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1399:12)
    at http://localhost:8080/Specs/Scene/ScreenSpaceCameraControllerSpec.js:1101:37
    at Object.<anonymous> (http://localhost:8080/Specs/spec-main.js:137:30)
    at attemptAsync (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1793:24)
    at QueueRunner.run (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1746:26)
```

File: Source/Scene/ScreenSpaceCameraController.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ define([
          * @type {Number}
          * @default 20.0
          */
-        this.minimumZoomDistance = 20.0;
+        this.minimumZoomDistance = 1.0;
```
Comment (pjcozzi) [inline Source/Scene/ScreenSpaceCameraController.js]: Mention this in CHANGES.md.

**Comment by bagnell** [comment]
@pjcozzi The tests are fixed. This is ready.

**Comment by pjcozzi** [comment]
Awesome.

**Pull request merged**